app/messages/emq/publishing_job.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require 'ostruct'
+
+module Emq
+  # This class should be responsible for publishing messages to the EMQ which are validated
+  # against an Avro schema stored in the RedPanda registry before being sent
+  class PublishingJob
+    attr_reader :bunny_config
+
+    # Initialize the publishing job with the bunny configuration
+    def initialize
+      # Load the bunny configuration from the Rails configuration and convert it to an OpenStruct
+      @bunny_config = PublishingJob.deep_open_struct(Rails.configuration.bunny)
+    end
+
+    # Publish a message to the EMQ
+    # @param [Object] objects the object or objects to publish
+    # @param [Object] create_messsage_configuration the configuration to construct
+    #                                               the message to publish from the given object(s)
+    # @param [String] schema_key the key of the schema to validate the message against
+
+    def publish(objects, create_messsage_configuration, schema_key) # rubocop:disable Metrics/AbcSize,Metrics/MethodLength
+      # Check if the schema_key exists in the subjects hash and return early if it does not
+      schema = bunny_config.amqp.schemas.subjects[schema_key]
+      return if schema.nil?
+
+      # Get the subject and version from the schema and return early if either is nil
+      subject = bunny_config.amqp.schemas.subjects[schema_key].subject
+      version = bunny_config.amqp.schemas.subjects[schema_key].version
+      return if subject.nil? || version.nil?
+
+      # Create a validator and sender for the subject and version
```

Comment:
Looks like the `Emq::Validator` class validates the message, doesn't it? `subject` and `version` are validated before in line 31? Do you think it's better to be renamed to something like `Emq::MessageValidator`?

Reply:
Sure, will do 👍

Reply:
I can see why you've suggested this because it may look like you're validating the subject and version.  I wonder whether a better solution is to not pass in the subject, version and schema registry and instead have a different module or class deal with getting the schema.  Then pass the schema to the validator's constructor.

Reply:
My fault:  I was focused on validation rather than the real job all the time. Renamed to encoder.